src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -524,75 +526,69 @@ export const handleStreamEncryptedResponses: RequestHandler<
  */
 export const handleGetEncryptedResponse: RequestHandler<
   { formId: string },
-  unknown,
+  EncryptedSubmissionDto | ErrorDto,
   unknown,
   { submissionId: string }
 > = async (req, res) => {
+  const sessionUserId = (req.session as Express.AuthedSession).user._id
   const { submissionId } = req.query
   const { formId } = req.params
 
-  const logMeta = {
-    action: 'handleGetEncryptedResponse',
-    submissionId,
-    formId,
-  }
-
-  // Step 1: Retrieve submission.
-  const submissionResult = await getEncryptedSubmissionData(
-    formId,
-    submissionId,
-  )
-
-  if (submissionResult.isErr()) {
-    logger.error({
-      message: 'Failure retrieving encrypted submission from database',
-      meta: logMeta,
-      error: submissionResult.error,
-    })
-
-    const { statusCode, errorMessage } = mapRouteError(submissionResult.error)
-    return res.status(statusCode).json({
-      message: errorMessage,
-    })
-  }
+  return (
+    // Step 1: Retrieve logged in user.
+    getPopulatedUserById(sessionUserId)
```

Comment:
think i need context here; why is this checked now when it wasn't previously ah

Reply:
if you look at the changes in `admin-form.routes`, there used to be a function called `authEncryptedResponseAccess` which did this

Reply:
if you look at the changes in `admin-form.routes`, there used to be a function called `authEncryptedResponseAccess` which did this